### PR TITLE
Restyle mobile nav trigger and address bugs related to its "stickiness"

### PR DESCRIPTION
**Pull request by erikphansen**
Blocked until this change to formation is merged and published to npm https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/581

## Description
This restyles the appearance of the mobile sub-navigation trigger button (https://github.com/department-of-veterans-affairs/va.gov-team/issues/13325) and fixes a handful of bugs (https://github.com/department-of-veterans-affairs/va.gov-team/issues/8256).

## Testing done
Local

## Screenshots

https://user-images.githubusercontent.com/20728956/105220039-01a11080-5b0c-11eb-9231-16f87a7100d4.mp4

## Acceptance criteria
- [ ]

## Definition of done
- [ ] Events are logged appropriately
- [ ] Documentation has been updated, if applicable
- [x] A link has been provided to the originating GitHub issue (or connected to it via ZenHub)
- [x] No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs

### CODE REVIEW

**Comment by sshein** [comment]
I think this looks great. excited to get this rolled in! The blue bar was completely overlooked when I ran testing for mobile month

File: src/site/components/navigation-sidebar-trigger.html [hunk 1]
```diff
@@ -8,37 +8,104 @@
...
 {% endcomment %}
-<button type="button" class="va-btn-sidebarnav-trigger vads-u-width--full"
-  aria-controls="va-detailpage-sidebar" id="sidebar-nav-trigger">
-  <span>
-    <b>More in this section</b>
-    {% include "src/site/assets/img/arrow-right-white.svg" %}
-  </span>
-</button>
-
+<div class="va-btn-sidebarnav-trigger" id="sidebar-nav-trigger">
```
Comment (timwright12) [inline src/site/components/navigation-sidebar-trigger.html:11]: Does the movement of this class and ID change which element the click event is bound to? Just want to make sure we're not binding a click to a div and it's still on the button element to keep this keyboard accessible. https://www.w3.org/WAI/WCAG21/quickref/#keyboard

Comment (erikphansen) [reply src/site/components/navigation-sidebar-trigger.html:11]: no the `button` is still the thing that's clickable and using the keyboard to interact with this still works great 👍  This is the corresponding change https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/581/files#diff-058155fb4dee49e02a161d9d2c6f3e45d3e8dc90c87a77084f9eb5ab35178bdbR153-R155

Comment (timwright12) [reply src/site/components/navigation-sidebar-trigger.html:11]: wonderful, thank you

**Review by timwright12 — APPROVED** [review]

**Comment by rtwell** [comment]
lookin' good to me, thanks for the hard work! 👍

**Pull request merged**